File: generate-imports.py
import glob
import os

priority = {
    'woff2': 1,
    'woff': 2,
    'otf': 3,
    'ttf': 4
}

tags = []
for f in glob.glob('./Ogma3/wwwroot/fonts/*'):
    path = os.path.normpath(f)
    path_els = path.split('\\')
    ext = path_els[-1].split('.')[-1]
    tag = f"url('/fonts/{path_els[-1]}') format('{ext}'),"
    tags.append((tag, priority[ext]))

tags.sort(key=lambda x: x[1])
tags.sort(key=lambda x: x[0], reverse=True)
for t in tags:
    print(t[0])

File: Tools/generate-imports.ts
const priorities= new Map<string, number>([
	['woff2', 1],
	['woff', 2],
	['otf', 3],
	['ttf', 4]
]);

let tags: { tag: string, priority: number }[] = [];

const path = new URL("../Ogma3/wwwroot/fonts/", import.meta.url);
for await (const f of Deno.readDir(path)) {
	const name = f.name.split('.')[1];
	tags.push({ tag: f.name, priority: priorities.get(name) ?? 0 });
}

tags.sort((a, b) => a.priority - b.priority);
tags.sort((a,b) => (a.tag > b.tag) ? 1 : ((b.tag > a.tag) ? -1 : 0)).reverse();

for (const t of tags) {
	console.log(`url('/fonts/${t.tag}') format('${t.tag.split('.')[1]}'),`);
}